### DISCUSSION TITLE
What is the best setup to use Quarto with the LanguageTool?

### DISCUSSION THREAD

**Discussion by Martin15135215** [quarto-dev/quarto-cli#5822] [Q&A]
### Description

Since LanguageTool also offers an add-on for browsers, I thought that Jupyter Lab is the easiest way to use Quarto and the advantages of LanguageTool.

Are there any other ideas from the community? :)

**Comment by JOduMonT** [comment]
Hi,
I use Quarto and LanguageTool in VS-Code. You need the 2 plugins since qmd are markdown file.

**Reply by Martin15135215** [reply]
Thanks for the input, i will try that out !

**Reply by JOduMonT** [reply]
#### While it's far from perfect (for now; it only notices you that you have a mistake)
![image](https://github.com/quarto-dev/quarto-cli/assets/5204724/3dcd9aca-fa5d-4639-a9eb-b2ef57d607fc)

#### and after installing the 2 plugins, you have to edit setting.json
![image](https://github.com/quarto-dev/quarto-cli/assets/5204724/93dcb17d-1747-4cab-bbb3-84b4803a7f91)

### it is possible to vote for a full integration
https://github.com/valentjn/vscode-ltex/issues/699

**Comment by mcanouil** [comment]
For whom may read this thread.

"[LanguageTool](https://languagetool.org/?force_language=1)" is a multilingual spelling, style, and grammar checker that helps correct or paraphrase texts.

A Visual Studio Code extension exists (not official): https://marketplace.visualstudio.com/items?itemName=adamvoss.vscode-languagetool

Quarto VSCode extension: https://marketplace.visualstudio.com/items?itemName=quarto.quarto

In addition Code Spell Checker VSCode extension: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker


Note: @Martin15135215 it's __Quarto__ not __Quatro__ 😉

**Reply by Martin15135215** [reply]
Typo fixed :)

and the git repo for the Visual Studio Code extension for LanguageTool shown above is archived -> see https://github.com/languagetool-language-server/vscode-languagetool -> there are two successor to my knowledge

**Reply by mcanouil** [reply]
The repo is archived but the extension is still listed in the marketplace.

**Reply by ekatko1** [reply]
Code from the archived languagetool VSCode extension seems to have been given a second life in the [LTeX+ VSCode extension](https://github.com/ltex-plus/vscode-ltex-plus)
